### ISSUE TITLE
[Openbmc] python framework concurrent issue

### ISSUE THREAD

**Issue by robin2008** [xcat2/xcat-core#4922]
when issuing two openbmc command in the same time (on different terminal), we can find that one could finish, but the other terminal will hang there.
```
time rinv fake1 all
Mon Mar 12 03:15:58 2018 fake1: [openbmc_debug] login curl -k -c cjar -b cjar -X POST -H "Content-Type: application/json" https://10.6.3.100/login -d '{"data": ["root", "xxxxxx"]}'
Mon Mar 12 03:15:58 2018 fake1: [openbmc_debug] login 200 OK
Mon Mar 12 03:15:58 2018 fake1: [openbmc_debug] get_inventory_info curl -k -c cjar -b cjar -X GET -H "Content-Type: application/json" https://10.6.3.100/xyz/openbmc_project/inventory/enumerate
Mon Mar 12 03:16:03 2018 fake1: [openbmc_debug] get_inventory_info 200 OK
Mon Mar 12 03:16:03 2018 fake1: [openbmc_debug] list_firmware curl -k -c cjar -b cjar -X GET -H "Content-Type: application/json" https://10.6.3.100/xyz/openbmc_project/software/enumerate
Mon Mar 12 03:16:04 2018 fake1: [openbmc_debug] list_firmware 200 OK
fake1: BMC BuildDate :
fake1: BMC FieldReplaceable : 0
fake1: BMC Manufacturer : IBM
fake1: BMC Model :
fake1: BMC PartNumber : 01DH118
fake1: BMC Present : 1
fake1: BMC PrettyName : BMC PLANAR
fake1: BMC SerialNumber : 000000000000
fake1: BMC UUID :
fake1: CPU0 BuildDate : 1996-01-01 - 00:00:00

...

fake1: Host Firmware Product: -- additional info: skiboot-v5.9.4

real	0m7.591s
user	0m0.220s
sys	0m0.016s
```
After you issue the command again in the first complete terminal, the hang one will exit without result, and the new one will hang again.

```
time rinv fake1 all
Mon Mar 12 03:16:00 2018 fake1: [openbmc_debug] login curl -k -c cjar -b cjar -X POST -H "Content-Type: application/json" https://10.6.3.100/login -d '{"data": ["root", "xxxxxx"]}'

real	2m28.648s
user	0m0.156s
sys	0m0.026s
```

**Comment by robin2008** [comment]
This issue is caused by that python framework (agent.py) is using the same lock file an socket file for communication. So that when multiple agent.py,  it will cause mess.

When you hit the issue, we can find the PID by `ps -ef`, for example
```
[root@c910f03c05k20 agent]# ps -ef|grep xcatd
root      2231     1  0 03:09 ?        00:00:00 xcatd: SSL listener
root      2232  2231  0 03:09 ?        00:00:00 xcatd: DB Access
postgres  2233  7072  0 03:09 ?        00:00:00 postgres: xcatadm xcatdb 10.3.5.20(39372) idle
root      2234  2231  0 03:09 ?        00:00:00 xcatd: UDP listener
root      2235  2231  0 03:09 ?        00:00:00 xcatd: install monitor
root      2236  2234  0 03:09 ?        00:00:00 xcatd: Discovery worker
root      2237  2231  0 03:09 ?        00:00:00 xcatd: Command log writer
postgres  2242  7072  0 03:09 ?        00:00:00 postgres: xcatadm xcatdb 10.3.5.20(39374) idle
root      2351  2231  0 03:18 ?        00:00:00 xcatd SSL: rinv to fake1 for root@localhost
root      2352  2351  0 03:18 ?        00:00:00 xcatd SSL: rinv to fake1 for root@localhost: openbmc2 instance
root      2383 29940  0 03:26 pts/0    00:00:00 grep --color=auto xcatd
[root@c910f03c05k20 agent]# ps -ef|grep agent
root      2354  2352  0 03:18 ?        00:00:00 python /opt/xcat/lib/python/agent/agent.py
```

And then you can just kill the remaining agent.py process for a workaround.

**Comment by robin2008** [comment]
This issue is addressed by #4921
